### DISCUSSION TITLE
Type 'void' is not assignable to type '{ user: Partial<User>; }'

### DISCUSSION THREAD

**Discussion by MananDesai54** [TanStack/query#7073] [Q&A]
### Describe the bug

Here `interface MutationOptions<TData = unknown, TError = DefaultError, TVariables = void, TContext = unknown> {` the default type of `TVariables` is void it is wrong and not letting me to re-assign any other type like here `{ user: Partial<User>; }` it should be `unknown`

### Your minimal, reproducible example

Find for Type `MutationOptions`

### Steps to reproduce

Find for Type `MutationOptions`

### Expected behavior

default type for `TVariables` should be `unknown`

### How often does this bug happen?

Everytime giving type for mutationFn variables type

### Screenshots or Videos

![image](https://github.com/TanStack/query/assets/52067783/6331a069-23bc-4897-91fa-7654cc683dc2)

### Platform

- OS : Manjaro Linux
- Browser: Brave

### Tanstack Query adapter

react-query

### TanStack Query version

v5.25.0

### TypeScript version

v5

**Comment by TkDodo** [comment]
show a typescript playground please

**Comment by TkDodo** [comment]
`TVariables` is `void` so that you can have mutations that don't take parameters. This is very much on purpose. Moving to discussion.

**Reply by MananDesai54** [reply]
But then how can I provide Params to mutationFn ? As we used to do in v4
